allennlp/training/trainer.py

```diff
@@ -409,6 +409,11 @@ def __init__(
         self._tensorboard.get_batch_num_total = lambda: self._batch_num_total
         self._tensorboard.enable_activation_logging(self.model)
 
+        # Only display reg_loss if the model's configuration has regularization.
+        self._show_metrics = ["loss", "reg_loss"]
+        if self.model.get_regularization_penalty() == 0.0:
```

Comment:
It's likely that we have to detect whether a regularization loss is configured by looking whether the number is `0`. But it would be better if we make that decision based on whether it's configured, not based on whether the result is zero. That way we can show it, for example, when it's misconfigured.

Reply:
I believe the `get_regularization_penalty()` function returns a 0 if the model is configured to not have regularization. This is what the docstring for that function says: `Returns 0 if the model was not configured to use regularization.`